modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/java/parser/ResourceOpenAPIParser.java

```diff
@@ -146,13 +146,54 @@ public static String getMethodAnnotations(
 				sb.append("description=\"");
 				sb.append(operation.getDescription());
 				sb.append("\"");
+
+				if (javaMethodSignature.getRequestBodyMediaTypes(
+					).contains(
+						"multipart/form-data"
+					)) {
+
+					sb.append(", requestBody = ");
+					sb.append("@io.swagger.v3.oas.annotations.parameters.");
+					sb.append("RequestBody(content = @io.swagger.v3.oas.");
+					sb.append("annotations.media.Content( mediaType = ");
+					sb.append("\"multipart/form-data\", schema = @io.swagger.");
+					sb.append("v3.oas.annotations.media.Schema( ");
+					sb.append("implementation = _");
+					sb.append(
+						StringUtil.upperCaseFirstLetter(
+							operation.getOperationId()));
+					sb.append("RequestBody.class)))");
+				}
 			}
 
 			sb.append(")");
 
 			methodAnnotations.add(sb.toString());
 		}
 
+		if ((operation.getDescription() == null) &&
+			javaMethodSignature.getRequestBodyMediaTypes(
+			).contains(
+				"multipart/form-data"
+			) && !operation.isDeprecated() &&
+			(getMultipartBodySchemas(javaMethodSignature) != null)) {
```

Comment:
I think that can simplify this condition
```java
javaMethodSignature.getRequestBodyMediaTypes(
			).contains(
				"multipart/form-data"
			) 
```
inside of `getMultipartBodySchemas`. something like:

```java
public static Map<String, Schema> getMultipartBodySchemas(
		JavaMethodSignature javaMethodSignature) {

		Operation operation = javaMethodSignature.getOperation();

		RequestBody requestBody = operation.getRequestBody();

		Map<String, Content> contentMap = requestBody.getContent();

		Content multipartContent = contentMap.get("multipart/form-data");
                
                if (content!=null) {
                           Schema schema = multipartContent.getSchema();

		          return schema.getPropertySchemas();

                }
                

		return null;
	}
```